examples/sep24/sep24.ts

```diff
@@ -0,0 +1,207 @@
+import axios from "axios";
+import readline from "readline";
+
+import {
+  walletSdk,
+  Anchor,
+  SigningKeypair,
+  Types,
+  IssuedAssetId,
+} from "@stellar/typescript-wallet-sdk";
+import { Memo, MemoText } from "stellar-sdk";
+
+const wallet = walletSdk.Wallet.TestNet();
+const stellar = wallet.stellar();
+const anchor = wallet.anchor({ homeDomain: "testanchor.stellar.org" });
+const account = stellar.account();
+
+let kp: SigningKeypair;
+
+const runSep24 = async () => {
+  const asset = new IssuedAssetId(
+    "USDC",
+    "GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5",
+  );
+
+  // Create Account
+
+  const createAccount = async () => {
+    console.log("creating account ...");
+    kp = account.createKeypair();
+    console.log(`kp: \n${kp.publicKey}\n${kp.secretKey}`);
+    await axios.get("https://friendbot.stellar.org/?addr=" + kp.publicKey);
+
+    const txBuilder = await stellar.transaction({
+      sourceAddress: kp,
+      baseFee: 1000,
+    });
+    const tx = txBuilder.addAssetSupport(asset).build();
+    kp.sign(tx);
+    const success = await stellar.submitTransaction(tx);
+    if (!success) {
+      throw new Error("adding trustline failed");
+    }
+  };
+
+  await createAccount();
+  await runDeposit(anchor, kp);
+  await runDepositWatcher(anchor);
+
+  while (!depositDone) {
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  }
+
+  const ans = await askQuestion("Do you want to start withdrawal? (y/n)");
+  if (ans !== "y") {
+    process.exit(0);
+  }
+
+  await runWithdraw(anchor, kp);
+  await runWithdrawWatcher(anchor, kp);
+};
+runSep24();
+
+// Create Deposit
+let authTokenDeposit: string;
+export const runDeposit = async (anchor: Anchor, kp: SigningKeypair) => {
+  console.log("\ncreating deposit ...");
+  const auth = await anchor.sep10();
+  authTokenDeposit = await auth.authenticate({ accountKp: kp });
+
+  const assetCode = "USDC";
+  const resp = await anchor.sep24().deposit({
+    assetCode,
+    authToken: authTokenDeposit,
+    lang: "en-US",
+    destinationMemo: new Memo(MemoText, "test-memo"),
+    extraFields: {
+      wallet_name: "Test Wallet",
+      wallet_url: "https://stellar.org/",
+    },
+  });
+
+  console.log("Open url:\n", resp.url);
+};
+
+// Watch Deposit
+export let depositDone = false;
+export const runDepositWatcher = (anchor: Anchor) => {
+  console.log("\nstarting watcher ...");
+
+  let stop: Types.WatcherStopFunction;
+  const onMessage = (m: Types.AnchorTransaction) => {
+    console.log({ m });
+    if (m.status === "completed") {
```

Comment:
Don't we have enum for statuses?

Reply:
https://github.com/stellar/typescript-wallet-sdk/pull/58/commits/3d0558484d2e6ee2a54b2ed2836c999995ef28de